src/huutonet/vahti.rs

```diff
@@ -0,0 +1,106 @@
+use std::sync::LazyLock;
+
+use async_trait::async_trait;
+use regex::Regex;
+
+pub static HUUTONET_REGEX: LazyLock<Regex> =
+    LazyLock::new(|| Regex::new(r"^https://(www\.)?huuto\.net/haku?.*$").unwrap());
+
+use super::api::{is_valid_url, vahti_to_api};
+use super::parse::api_parse_after;
+use crate::error::Error;
+use crate::models::DbVahti;
+use crate::vahti::{Vahti, VahtiItem};
+use crate::Database;
+
+#[derive(Debug, Clone)]
+pub struct HuutonetVahti {
+    pub id: i32,
+    pub url: String,
+    pub user_id: u64,
+    pub last_updated: i64,
+    pub site_id: i32,
+    pub delivery_method: i32,
+}
+
+#[async_trait]
+impl Vahti for HuutonetVahti {
+    async fn update(&mut self, db: &Database) -> Result<Vec<VahtiItem>, Error> {
+        debug!("Updating {}", self.url);
+
+        let res = reqwest::get(vahti_to_api(&self.url))
+            .await?
+            .text()
+            .await?
+            .to_string();
+
+        let ret = api_parse_after(&res, self.last_updated)?
+            .iter()
+            .cloned()
```

Comment:
often, .iter().cloned() can be replaced with .into_iter() for more handsomeness. Does that apply here?